src/coreComponents/physicsSolvers/SolverStatistics.hpp

```diff
@@ -22,176 +22,381 @@
+    /// String key for the discarded number of nonlinear iterations
+    static constexpr char const * numDiscardedNonlinearIterationsString() { return "numDiscardedNonlinearIterations"; }
+    /// String key for the discarded number of linear iterations
+    static constexpr char const * numDiscardedLinearIterationsString() { return "numDiscardedLinearIterations"; }
+  };
 
   /**
-   * @brief Initialize the counters used for an individual time step
+   * @brief Set the csv state output
+   * @param state The csv state
    */
-  void initializeTimeStepStatistics();
+  void setCSVOutputState( bool state )
+  { m_csvOutputEnabled = state; }
 
   /**
-   * @brief Tell the solverStatistics that we are doing a nonlinear iteration
-   * @param[in] numLinearIterations the number of linear iterations done by the linear solver
-   * @detail This function is well suited for Newton's method, or for single-physics solvers in sequential schemes
+   * @brief Set the log state output
+   * @param state The log state
    */
-  void logNonlinearIteration( integer const numLinearIterations );
+  void setLogOutputState( bool state )
+  { m_logOutput = state; }
+
 
   /**
-   * @brief Tell the solverStatistics that we are doing a nonlinear iteration
-   * @detail This function is well suited for the outer loop in sequential schemes
+   * @brief Reset the counters used for an individual time step
    */
-  void logNonlinearIteration();
+  void resetCurrentTimeStepStatistics();
 
   /**
-   * @brief Tell the solverStatistics that we are doing an outer loop iteration
+   * @brief Tell the solverStatistics that we have done a configuration iteration
    */
-  void logOuterLoopIteration();
+  void incrementConfigIteration();
 
   /**
-   * @brief Tell the solverStatistics that there is a time step cut
+   * @brief Tell the solverStatistics that we have done a newton iteration
+   * @param[in] numLinearIterations the number of linear iterations done by the linear solver
+   * @detail This function is well suited for Newton's method, or for single-physics solvers in sequential schemes
    */
-  void logTimeStepCut();
+  void updateNonlinearIteration( integer const numLinearIterations );
 
   /**
-   * @brief Save the statistics for the individual time step and increment the cumulative stats
+   * @brief Accumulate the setupTime & solveTime result over each newton iteration
+   * @param setupTime The linear setup time
+   * @param solveTime The linear solve time
    */
-  void saveTimeStepStatistics();
+  void accumulateSolverLinearTime( real64 setupTime, real64 solveTime );
 
   /**
-   * @brief Output the cumulative statistics to the terminal
+   * @brief Reset the setupTime & solveTime to 0 at the end of each cycle
    */
-  void outputStatistics() const;
+  void resetSolverLinearTime();
 
   /**
-   * @return Number of time steps
+   * @brief Tell the solverStatistics that we cut the time step and we increment the cumulative counters for discarded timesteps
```

Comment:
Maybe, to explain the timestep cut workflow.
```suggestion
   * @brief Tell the solverStatistics that we cut the time step and we increment the cumulative counters for discarded timesteps
   *        Assume that we restart computing the last timestep cut we are computing (it did not converge).
```